test/flow_runner/simulator_test.exs

```diff
@@ -1028,4 +1028,27 @@ defmodule FlowRunner.SimulatorTest do
     # does NOT exists in the context variables
     assert get_in(sim.context.vars, ["variables", "platform"]) == nil
   end
+
+  test "simulator with wait block" do
+    sim = Simulator.new(read_floip!("wait"))
+
+    {:end, _sim, outputs} = Simulator.start(sim)
+
+    # Should have two messages: wait debug message and the actual message
+    assert length(outputs) == 2
+
+    # First output should be the wait block debug message
+    [wait_output, message_output] = outputs
+
+    assert {:message, wait_fields} = wait_output
+    [wait_text] = get_in(wait_fields, [:text])
+    assert wait_text.content_type == "TEXT"
+    assert wait_text.value =~ "Paused execution for 1 second(s)"
```

Comment:
Nit: Can we match the entire text, including [DEBUG]?
I think it’s useful to keep it explicit so no one accidentally removes it, it makes clear that this is just debugging output intended only for simulator runs.

Reply:
Done in [56286c9](https://github.com/turnhub/flow_runner/pull/271/commits/56286c9eac01ad935fc5cb109d0ceb720be2a309).